Extend/Controller/Admin/PaymentMain.php

```diff
@@ -215,7 +225,65 @@ private function _validateRequestParameters($aParams)
             return $oTransactionService->checkCredentials();
         }
 
-        return false;
+        // handle the case of different configuration per currency (currently only Payolution)
+        // there it's only checked if config values are entered on the frontend but no validation is done on the backend
+        return $this->_checkCurrencyConfigFields($aParams);
+    }
+
+    /**
+     * Checks if the config fields for all selected allowed currencies have been set.
+     *
+     * @param array $aParams
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    private function _checkCurrencyConfigFields($aParams)
+    {
+        $aOldCurrencyValue = $this->_getPaymentMethod()->oxpayments__allowed_currencies->value;
+        $aNewCurrencyValue = $aParams['oxpayments__allowed_currencies'];
+
+        foreach ($aNewCurrencyValue as $sCurrency) {
+            // it is only necessary to check this currency at this point if it was already saved before
+            if (!in_array($sCurrency, $aOldCurrencyValue)) {
+                continue;
+            }
+
+            if (!$this->_validateRequiredCurrencyFields($aParams, $sCurrency)) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Checks that all the required fields for the selected currencies have been set.
+     *
+     * @param array  $aParams
+     * @param string $sCurrency
+     *
+     * @return bool
+     *
+     * @since 1.2.0
+     */
+    private function _validateRequiredCurrencyFields($aParams, $sCurrency)
+    {
+        $aRequiredFields = [
+            'oxpayments__httpuser_',
+            'oxpayments__httppass_',
+        ];
+
+        foreach ($aRequiredFields as $sFieldNamePrefix) {
+            $sFieldName = $sFieldNamePrefix . strtolower($sCurrency);
+
+            if (!isset($aParams[$sFieldName]) || empty($aParams[$sFieldName])) {
```

Comment:
The `isset` check is not needed – `empty` catches nonexistent keys as well.

Reply:
👍